### PR TITLE
samples: openthread: enable USB transport for CLI sample

### PR DESCRIPTION
**Pull request by lmaciejonczyk**
Add new overlay configuration file which enables USB transport for the CLI
sample. Extend Thread CLI sample support for nrf52840 dongle.

Depends on:

- [x] #4121

### CODE REVIEW

**Review by rlubos — APPROVED** [review]

File: samples/openthread/cli/README.rst [hunk 1]
```diff
@@ -79,7 +86,7 @@ The sample supports the following development kits for testing the network statu
 
 .. table-from-rows:: /includes/sample_board_rows.txt
    :header: heading
-   :rows: nrf5340dk_nrf5340_cpuapp, nrf52840dk_nrf52840, nrf52833dk_nrf52833, nrf21540dk_nrf52840
+   :rows: nrf5340dk_nrf5340_cpuapp, nrf52840dk_nrf52840, nrf52840dongle_nrf52840, nrf52833dk_nrf52833, nrf21540dk_nrf52840
```
Comment (edmont) [inline samples/openthread/cli/README.rst:89]: Should we add it here as well? https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/app_boards.html?highlight=nrf52840dongle_nrf52840#boards-included-in-sdk-nrf

File: samples/openthread/cli/overlay-usb.conf [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+#
+
+CONFIG_SHELL_BACKEND_SERIAL_INIT_PRIORITY=51
+
+CONFIG_UART_LINE_CTRL=y
+CONFIG_UART_SHELL_ON_DEV_NAME="CDC_ACM_0"
+
+CONFIG_USB=y
+CONFIG_USB_DEVICE_MANUFACTURER="Nordic Semiconductor ASA"
+CONFIG_USB_DEVICE_PRODUCT="Thread Cli"
```
Comment (edmont) [inline samples/openthread/cli/overlay-usb.conf:14]:
```suggestion
CONFIG_USB_DEVICE_PRODUCT="Thread CLI"
```

**Review by edmont — COMMENTED** [review]
Should we also update our docs on how to flash the dongle? Since `west flash` won't handle it. Maybe just a link to here: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/zephyr/boards/arm/nrf52840dongle_nrf52840/doc/index.html

**Comment by lmaciejonczyk** [comment]
> Should we also update our docs on how to flash the dongle? Since `west flash` won't handle it. Maybe just a link to here: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/zephyr/boards/arm/nrf52840dongle_nrf52840/doc/index.html

I agree. What more It looks this is a missing part also for other samples which support nRF52840 dongle as well. I think It should be done as general fix, maybe in `gs_programming.rst`?

**Review by ru-fu — CHANGES_REQUESTED** [review]
I'll add a commit.

**Comment by ru-fu** [comment]
I added two commits now - one for the language review and one to add a note to the programming instructions.

**Review by edmont — APPROVED** [review]
LGTM now.

@ru-fu do you mind squashing the language review commit?

**Review by ru-fu — APPROVED** [review]
Squashed and rebased.

**Pull request merged**